src/ImageSharp/Formats/Jpeg/JpegEncoder.cs

```diff
@@ -50,7 +56,21 @@ public Task EncodeAsync<TPixel>(Image<TPixel> image, Stream stream, Cancellation
             where TPixel : unmanaged, IPixel<TPixel>
         {
             var encoder = new JpegEncoderCore(this);
+            this.EnrichColorType<TPixel>();
             return encoder.EncodeAsync(image, stream, cancellationToken);
         }
+
+        /// <summary>
+        /// If ColorType was not set, set it based on the given <typeparamref name="TPixel"/>.
+        /// </summary>
+        private void EnrichColorType<TPixel>()
```

Comment:
`InitializeColorType` would be a better name for this method.

Reply:
Renamed it as you suggested